vars/unitTest.groovy

```diff
@@ -87,13 +87,10 @@ def call(Map config = [:]) {
     stashes.add("${target_compiler}-install")
     stashes.add("${target_compiler}-build-vars")
   }
-  
-  if (stage_info['valgrind']) {
-    test_script = test_script.trim() + " " + stage_info['valgrind']
-  }
 
   Map params = [:]
   params['stashes'] = stashes
+  params['with_valgrind'] = stage_info['valgrind']
   params['script'] = "SSH_KEY_ARGS=${env.SSH_KEY_ARGS} " +
                      "NODELIST=${nodelist} " +
                      test_script
```

Comment:
what is test_script here?   Could you just do "WITH_VALGRIND=" + stage_info['valgrind'] " + test_script ?

Reply:
We cannot make sure how test_script will change in the future, that's why @brianjmurrell suggested the environment variable. I set up the variable in `runTest.groovy` instead of `unitTest.groovy` and it works now.

Reply:
The stage name in Jenkins in Jenkins for non-matrix jobs should be unique and have the keywords in it to set the defaults used in the block of the stage.

This allows cut/paste and no edits except for the stage name to setup the actions for the stage.

When matrix jobs start working, the matrix will set environment variables directly instead of parsing the stage name, which will be more generic.